### PR TITLE
Fix Ubuntu:14.04 install instructions.

### PR DESCRIPTION
**Pull request by coryan**
I thought I had tested these, but obviously I did not because they did
not work :disappointed: I guess that is why we have CI builds.

This time I did run the build script.  I am looking forward to April when
Ubuntu:14.04 goes EOL and we can remove all these hacks.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2056)

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -501,11 +501,21 @@ cd $HOME/Downloads
 wget -q https://www.openssl.org/source/openssl-1.0.2n.tar.gz
 tar xf openssl-1.0.2n.tar.gz
 cd $HOME/Downloads/openssl-1.0.2n
-./Configure --prefix=/usr/local --openssldir=/usr/local linux-x86_64
+./config --shared
```
Comment (devbww) [inline INSTALL.md]: Is /usr/local/ssl the default prefix?  Should we be explicit anyway?

Comment (remyabel) [reply INSTALL.md]: If you specify nothing, both `prefix` and `openssldir` will be set to `/usr/local/ssl`. If you specify only `prefix`, it becomes `/usr/local/` and `/usr/local/ssl` respectively. If you specify only `openssldir`, it becomes `/usr/local/` and `/usr/local` respectively. If you specify both, same behavior. Personally I find this behavior to be kind of unintuitive.

Comment (devbww) [reply INSTALL.md]: Would not `./config --openssldir=/usr/local/ssl --shared` do the right thing?

File: INSTALL.md [hunk 2]
```diff
@@ -506,8 +506,15 @@ make -j $(nproc)
 sudo make install
 ```
 
-We also need to configure CMake to find the version of OpenSSL we just
-installed:
+Note that by default OpenSSL installs itself in `/usr/local/ssl`. Installing
+on a more conventional location, such as `/usr/local` or `/usr`, can break
```
Comment (devbww) [inline INSTALL.md]: s/on/in/

File: INSTALL.md (hunk 1 above)
Comment (coryan) [reply INSTALL.md]: I am just going to describe what happens and why the default is a good location in this case. Seems like explicitly setting the destination is involved and more than I feel comfortable describing.

Did I mention that I cannot wait for Ubuntu:14.04 to be EOL?

File: INSTALL.md (hunk 2 above)
Comment (coryan) [reply INSTALL.md]: Done.

**Comment by coryan** [comment]
PTAL.

**Comment by coryan** [comment]
I am going to merge this because at least it will fix the CI builds.  Do not hesitate to request additional fixes, I will just create a second PR.

**Pull request merged**